src/Features/LanguageServer/Protocol/Handler/Completion/AbstractLspCompletionResultCreationService.cs

```diff
@@ -33,7 +33,7 @@ internal abstract class AbstractLspCompletionResultCreationService : ILspComplet
             Document document,
             int position,
             CompletionCapabilityHelper capabilityHelper,
-            CompletionList list, bool isIncomplete, long resultId,
+            CompletionList list, bool isIncomplete, long resultId, long documentId,
```

Comment:
i'm confused as to waht this 'long' represents.  docIds are a strong type, that aren't encoded as a long.

Reply:
Same as resultId, it's a cache index. The entire completion list as well as the document identifier are stored in a "resolve cache" so that only the IDs need to be added to each item's request data. And the resolve handler looks up the actual data again based on these IDs. This is done to minimize the data being passed around.